zebra-consensus/src/transaction/tests.rs

```diff
@@ -241,3 +241,45 @@ async fn v5_transaction_is_accepted_after_nu5_activation() {
         assert_eq!(result, Ok(expected_hash));
     }
 }
+
+#[tokio::test]
+// TODO: Remove `should_panic` once the NU5 activation heights for testnet and mainnet have been
+// defined.
+#[should_panic]
+async fn transaction_is_rejected_based_on_script() {
+    let network = Network::Mainnet;
+    let blocks = zebra_test::vectors::MAINNET_BLOCKS.iter();
+
+    let state_service = service_fn(|_| async {
+        Err(Box::new(io::Error::new(
+            io::ErrorKind::Other,
+            "Pretending the UTXO was not found",
+        )) as BoxError)
+    });
+
+    let script_verifier = script::Verifier::new(state_service);
+    let verifier = Verifier::new(network, script_verifier);
+
+    let transaction = fake_v5_transactions_for_network(network, blocks)
+        .rev()
+        .find(|transaction| {
+            !transaction.is_coinbase()
+                && transaction.inputs().len() > 0
```

Comment:
Clippy says:
```
267 |                 && transaction.inputs().len() > 0
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: using `!is_empty` is clearer and more explicit: `!transaction.inputs().is_empty()`
    |
    = note: `-D clippy::len-zero` implied by `-D warnings`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#len_zero
```

It might be helpful to always run clippy before you push. You could use a git hook, or turn it on by default in your editor or build checks. I run:
```sh
cargo clippy --all-targets --all-features && cargo test
```

Using these scripts, which run clippy and test on the subproject I'm modifying right now, and filter out useless log entries:
https://github.com/teor2345/dotfiles/blob/main/bin/cargo-test-subproject
https://github.com/teor2345/dotfiles/blob/main/bin/cargo-watch-subproject

(I have to pick which subproject I'm using, it's not automatic.)

Reply:
Oops, done :+1: